app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_posts_like_counter
+    User.update(PostsCounter: 1)
+  end
+
```

Comment:
### :red_circle: **[MANDATORY CHANGE]**

- [ ] It would be nice if you make your update_posts_like_counter as you can see below because the method you make is updating all users posts counter but it should only update the posts counter of that respective user who has created the post so it would be great if you make your method as below :+1:

```ruby
  def update_posts_like_counter
    author.increment!(:PostsCounter)
  end
```

### :yellow_circle: **[OPTIONAL BUT HIGHLY RECOMMENDED]**

- [ ] It would be nice if you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_posts_like_counter method every time a new post has been saved to the database, then make the update_posts_like_counter method private method :+1: You can see below what I am talking :+1:

```ruby
after_save :update_posts_like_counter

  private

  def update_posts_like_counter
    author.increment!(:PostsCounter)
  end
```